sdk/include/platform/ibex/platform-dma.hh

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#define MALLOC_QUOTA 0x100000
+#define TEST_NAME "Allocator"
+#include "tests.hh"
+
+#include <cheri.hh>
+#include <compartment-macros.h>
+#include <cstdint>
+#include <stdint.h>
+#include <utils.hh>
+
+#include <thread.h>
+#include <thread_pool.h>
+
+using thread_pool::async;
+DECLARE_AND_DEFINE_ALLOCATOR_CAPABILITY(dmaDriverHeap, 8144);
+
+namespace Ibex {
+    class GenericDMA {
+        private:
+
+        Timeout noWait{0};
```

Comment:
You probably don't want this here, allocating a new one on the stack when you use it will be faster than having a global.  It is also not sensibly reentrant - the field in `Timeout` that tracks the amount that's consumed is not updated atomically and so it will end up being nonsense if you use the same timeout from multiple threads.